Admin/Form/EventListener/AddEntityIdentifierFieldSubscriber.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Grossum\MenuBundle\Admin\Form\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\FormInterface;
+
+use Grossum\MenuBundle\Entity\BaseMenuItem;
+use Grossum\MenuBundle\Manager\MenuManager;
+
+class AddEntityIdentifierFieldSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var MenuManager
+     */
+    private $menuMaster;
+
+    /**
+     * @param MenuManager $menuMaster
+     */
+    public function __construct(MenuManager $menuMaster)
+    {
+        $this->menuMaster = $menuMaster;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'onPreSetData',
+            FormEvents::PRE_SUBMIT   => 'onPreSubmit',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function onPreSetData(FormEvent $event)
+    {
+        $form = $event->getForm();
+
+        /* @var $menuItem BaseMenuItem */
+        $menuItem = $event->getData();
+
+        if ($menuItem xor $menuItem->getEntityClass()) {
+            return;
+        }
+
+        $this->addEntityIdentificatorField($form, $menuItem->getEntityClass());
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function onPreSubmit(FormEvent $event)
+    {
+        $data = $event->getData();
+        $form = $event->getForm();
+
+        if (isset($data['entityClass']) xor $data['entityClass']) {
```

Comment:
There is something strange.
@cowboy4ik89 could you check this expression again?